### PR TITLE
Add inspect tool

### PR DESCRIPTION
**Pull request by jwilder**
Start of a lower-level file inspection tool.  This currently dumps
summary statistics for the shards, index and WAL that can be used to
understand the shape of the data is in the local shards.  This util
operates on the shards itself and not through the server and is intended
more for debugging/troubleshooting.

Sample output after run `influx_stress`:

```
Shards: 1, Indexes: 1, Databases: 1, Disk Size: 536870912, Series: 100000

DB           Measurement  Tags [#K/#V]   Fields [Name:Type]  Series
stress       cpu          2/100001       1 [value:float64]   100000
```

### CODE REVIEW

File: tsdb/meta.go [hunk 1]
```diff
@@ -309,8 +309,7 @@ type Measurement struct {
 	index      *DatabaseIndex
 
 	// in-memory index fields
-	seriesByID          map[uint64]*Series // lookup table for series by their id
-	measurement         *Measurement
```
Comment (otoolep) [inline tsdb/meta.go]: `measurement` unused?

File: cmd/inspect/main.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+					fields, err := codec.DecodeFieldsWithNames(value)
+					if err != nil {
+						fmt.Printf("Failed to decode values: %v", err)
+					}
+
+					for field, value := range fields {
+						fieldSummary = append(fieldSummary, fmt.Sprintf("%s:%T", field, value))
+					}
+					sort.Strings(fieldSummary)
+					tx.Rollback()
```
Comment (otoolep) [inline cmd/inspect/main.go]: So if `len(series)` is not > 0, the tx is never rolled back? Will it be rolled back on the next loop? I want to be sure everything will be released as you needed.

File: tsdb/meta.go (hunk 1 above)
Comment (jwilder) [inline tsdb/meta.go]: Oh. Yeah.  Unused but not part of this PR.  Will remove.

File: cmd/inspect/main.go (hunk 2 above)
Comment (jwilder) [inline cmd/inspect/main.go]: Should be outside of the block.  The process will exit and rollback the tx though.

File: tsdb/meta.go (hunk 1 above)
Comment (otoolep) [inline tsdb/meta.go]: I'm OK with it getting removed here, I just wanted to check it's what you want.

**Comment by toddboom** [comment]
:heart:

**Comment by corylanou** [comment]
Any reason not to merge this today?  +1

**Comment by beckettsean** [comment]
@jwilder I'd love to get an overview or even a tech talk for the support team so that we understand what we can do with this tool. Or is it too low level to provide useful data for us?

**Comment by jwilder** [comment]
It will be useful for support in the future but there is not much to it yet.  Just run `inspect` or `inspect -h` to see the (one) option.  The idea is to add support for the other file types we store to aid in troubleshooting storage related issues.  It might also be useful as a step for logging issues so that we can get more consistent data around DB size, cardinality, OS, versions, etc.. when issues are logged. (i.e someone could just run `inspect -p /var/lib/influxdb` and attach that to the issue to help us out).
